### PR TITLE
fixing pandas 1.6.0 and 2.0.0 breaking changes

### LINKED ISSUE #575: Daily run failure: Unit tests
The daily unit tests failed. See https://github.com/Quantco/glum/actions/runs/3232211996 for details.

### PR DESCRIPTION
**Pull request by MarcAntoineSchmidtQC**
In the development version of pandas, `get_dummies` generates boolean data, and when used in arithmetic operations, this becomes a numpy array with dtype python object.

Glum was actually not the problem. When using a pandas dataframe as inputs, tabmat takes care to cast everything to the right dtype. The problem was with the tests themselves where they would pandas dataframes with `.values` to acces the underlying numpy array.

Coincidentally, 2 days ago pandas released the 2.0.0 development version. Since we started seeing the problem before that, it seems to be happening with 1.6.0 and 2.0.0.

This PR fixes #575.

**TODO: remove daily CI on push**

Checklist
* [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Comment by jtilly** [comment]
The `get_dummies` dtype change was introduced in 1.6.0 and classified as bugfix, see https://github.com/pandas-dev/pandas/pull/48022#pullrequestreview-1111560850.

File: tests/glm/test_glm.py [hunk 1]
```diff
@@ -1827,7 +1827,7 @@ def test_dataframe_std_errors(regression_data, categorical, split, fit_intercept
         alpha=0, family="normal", fit_intercept=fit_intercept
     )
     mdl.fit(X=X, y=y)
-    X_sm = pd.get_dummies(X)
+    X_sm = pd.get_dummies(X).to_numpy(dtype=float)
     if fit_intercept:
```
Comment (jtilly) [inline tests/glm/test_glm.py:1831]: `get_dummies` also accepts a dtype argument.

**Review by jtilly — APPROVED** [review]

**Comment by bashtage** [comment]
I agree that this change is problematic.  It caused the canary build of statsmodels to also start failing.  I think this change unfortunately is going to cause a lot of problems, and it should seriously be considered for rolling back.  The `np.asarray(df.get_dummies())` returning object is a very hard to debug problem.  Even `df.get_dummies().to_numpy()` produces a not-useful dtype.

**Comment by bashtage** [comment]
Oops.  I should have made most of my case on the pandas tracker.

**Pull request merged**